tools/c7n_azure/c7n_azure/session.py

```diff
@@ -53,19 +55,23 @@
 class Session(object):
 
     def __init__(self, subscription_id=None, authorization_file=None,
-                 resource=constants.RESOURCE_ACTIVE_DIRECTORY):
+                 cloud_endpoints=AZURE_PUBLIC_CLOUD, auth_endpoint=DEFAULT_AUTH_ENDPOINT):
         """
         :param subscription_id: If provided overrides environment variables.
         :param authorization_file: Path to file populated from 'get_functions_auth_string'
-        :param resource: Resource endpoint for OAuth token.
+        :param cloud_endpoints: List of endpoints for specified Azure Cloud. Defaults to public.
+        :param auth_endpoint: Resource endpoint for OAuth token.
         """
 
         self._provider_cache = {}
         self.subscription_id_override = subscription_id
         self.credentials = None
         self.subscription_id = None
         self.tenant_id = None
-        self.resource_namespace = resource
+        self.endpoints = cloud_endpoints or AZURE_PUBLIC_CLOUD
```

Comment:
nit: `cloud_endpoints` is enough because `AZURE_PUBLIC_CLOUD` is a default value for the parameter

Reply:
When `get_session_factory` is called [here](https://github.com/cloud-custodian/cloud-custodian/blob/master/tools/c7n_azure/c7n_azure/provider.py#L43), it initially sets `cloud_endpoints=None`. So when the `resolve_auth_endpoint` is called in the constructor, it is called with empty cloud endpoints causing an error to be thrown.

Reply:
I'd try to look more into where/why this is happening, is the `partial` causing a problem?  It feels weird to have the constructor defensively try to solve problems with callers sending invalid parameters.   It looked like it was set correctly everywhere (or not passed) but I haven't run it :)